### ISSUE TITLE
Re-importing data not overwriting correctly in some cases

### ISSUE THREAD

**Issue by MikeStower** [QAAFI/REMS2020#225]

**Comment by MikeStower** [comment]
@jbrider I've looked into this - it seems that it's only treatments that are getting duplicated. The current behaviour is that all treatments in the spreadsheet are always added as new entities in the database, causing issues if the same dataset is imported multiple times.

If a data set is imported again, and the listed treatments have changed for an experiment, should the old set of treatments be completely replaced/removed by the new set? Or should any existing treatments be left as is, with any new treatments that are detected added?

**Comment by jbrider** [comment]
It should probably be replaced with a very definite warning - because it should remove all the old data as well shouldn't it?

**Comment by MikeStower** [comment]
Yes, it would be a cascade delete - so any plot data tied to that treatment would also be expunged. The data spreadsheet would have to be re-imported as well. How should the user be notified? Just through a popup box post-import?

**Comment by jbrider** [comment]
post import?
After their data is deleted?
Or do you mean after the data is initially read in?

**Comment by MikeStower** [comment]
I was initially thinking after the data was already deleted, but in hindsight that seems like poor design. I'm now thinking it should be possible to have a confirmation dialog popup during the import that asks the user to confirm the override.

**Comment by MikeStower** [comment]
Working on this in 32bb473e3abdf6cc0710ba2311513e49840b26d4.

Treatments no longer duplicate, but it seems some operations are still duplicating.

**Comment by MikeStower** [comment]
Working on this in 54c891eab89a2b54e04007be287d10a67560e837 
Still needs a bit more testing to make sure no bugs have crept in.

**Comment by jbrider** [comment]
Have you written any tests?

**Comment by MikeStower** [comment]
There's no proper unit testing, I was just going to manually test using the files I've got.

**Comment by MikeStower** [comment]
Working on this in 53ae0b54923cbb57e5aee01fbbceaacd31c5d56a
